Sources/Classes/RSMoEngageDestination.swift

```diff
@@ -137,15 +167,27 @@ extension RSMoEngageDestination {
         for (key, value) in traits {
             if value is String || value is NSNumber || value is Date {
                 switch key {
-                case RSKeys.Identify.Traits.email: MoEngage.sharedInstance().setUserEmailID(value as? String)
-                case RSKeys.Identify.Traits.name: MoEngage.sharedInstance().setUserName(value as? String)
-                case RSKeys.Identify.Traits.phone: MoEngage.sharedInstance().setUserMobileNo(value)
-                case RSKeys.Identify.Traits.firstName: MoEngage.sharedInstance().setUserAttribute(value, forKey: USER_ATTRIBUTE_USER_FIRST_NAME)
-                case RSKeys.Identify.Traits.lastName: MoEngage.sharedInstance().setUserLastName(value as? String)
-                case RSKeys.Identify.Traits.gender: MoEngage.sharedInstance().setUserAttribute(value, forKey: USER_ATTRIBUTE_USER_GENDER)
-                case RSKeys.Identify.Traits.birthday: identifyDateUserAttribute(value: value, key: USER_ATTRIBUTE_USER_BDAY)
-                case RSKeys.Identify.Traits.address: MoEngage.sharedInstance().setUserAttribute(value, forKey: RSKeys.Identify.Traits.address)
-                case RSKeys.Identify.Traits.age: MoEngage.sharedInstance().setUserAttribute(value, forKey: RSKeys.Identify.Traits.age)
+                case RSKeys.Identify.Traits.email:
+                    if let email = value as? String {
+                       MoEngageSDKAnalytics.sharedInstance.setEmailID(email)
+                    }
+                case RSKeys.Identify.Traits.name:
+                    if let name = value as? String {
+                        MoEngageSDKAnalytics.sharedInstance.setName(name)
+                    }
+                case RSKeys.Identify.Traits.phone:
+                    if let phone = value as? String {
+                       MoEngageSDKAnalytics.sharedInstance.setMobileNumber(phone)
+                    }
+                case RSKeys.Identify.Traits.firstName: MoEngageSDKAnalytics.sharedInstance.setUserAttribute(value, withAttributeName: RSKeys.Identify.Traits.firstName)
+                case RSKeys.Identify.Traits.lastName:
+                    if let lastName = value as? String {
+                       MoEngageSDKAnalytics.sharedInstance.setLastName(lastName)
+                    }
+                case RSKeys.Identify.Traits.gender: MoEngageSDKAnalytics.sharedInstance.setUserAttribute(value, withAttributeName: RSKeys.Identify.Traits.gender)
+                case RSKeys.Identify.Traits.birthday: identifyDateUserAttribute(value: value, key: RSKeys.Identify.Traits.birthday)
+                case RSKeys.Identify.Traits.address: MoEngageSDKAnalytics.sharedInstance.setUserAttribute(value, withAttributeName: RSKeys.Identify.Traits.address)
+                case RSKeys.Identify.Traits.age: MoEngageSDKAnalytics.sharedInstance.setUserAttribute(value, withAttributeName: RSKeys.Identify.Traits.age)
                 default: identifyDateUserAttribute(value: value, key: key)
```

Comment:
Split into multiple functions as per the value type & apply the optional binding.
e.x: The 'default' case handles the value as date.

Reply:
Done